File: DBGuardAPI/Data/DTOs/GuardDTOs/GuardChangeTransactionDTO.cs
using DBGuardAPI.Data.Enums;
using Sieve.Attributes;

namespace DBGuardAPI.Data.DTOs.GuardDTOs
{
    public class GuardChangeTransactionDTO
    {
        [Sieve(CanSort = true, CanFilter = true)]
        public int Id { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public DateTimeOffset Timestamp { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public int? GuardId { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public GuardState GuardState { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public GuardState PreviousGuardState { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public required string GuardQuery { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public GuardOperator GuardOperator { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public int GuardValue { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public int? DatabaseConnectionId { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public required string DatabaseConnectionEndpoint { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public required string DatabaseName { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public DatabaseEngine DatabaseConnectionEngine { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public required string DatabaseConnectionUsername { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public int? ResultValue { get; set; }
    }
}

File: DBGuardClient/src/app/interfaces/guard-change-transaction-dto.ts
import { DatabaseEngine } from "../enums/database-engines";
import { GuardOperator } from "../enums/guard-operator";
import { GuardState } from "../enums/guard-state";

export interface GuardChangeTransactionDTO {
    id: number,
    timestamp: Date,
    guardId?: number,
    guardState: GuardState,
    previousGuardState: GuardState,
    triggerQuery: string,
    triggerOperator: GuardOperator,
    triggerValue: number,
    triggeredValue?: number,
    databaseConnectionId?: number,
    databaseConnectionEndpoint: string,
    databaseName: string,
    databaseConnectionEngine: DatabaseEngine,
    databaseConnectionUsername?: string,
    message?: string
}
export interface GuardChangeItemDTO {
    month: string,
    guardState: GuardState,
    count: number
}